### ISSUE TITLE
Passing only --tls-cert-file produces cryptic error message

### ISSUE THREAD

**Issue by errordeveloper** [kubernetes/kubernetes#19310]
It says: `E0105 14:50:09.165099      10 server.go:633] Unable to listen for secure (open : no such file or directory); will try again.`

I've also had `--cert-dir` set at the time, not sure if that somehow triggered an edge case of sorts. I've only tested this with 1.1.3 binaries.

**Comment by AdoHe** [comment]
@errordeveloper actually from source code kube-apiserver will use cert dir only if both `--tls-cert-file` and `--tls-private-key-file` is empty so if you just pass one and left other empty, apiserver will not load files from cert dir.

**Comment by errordeveloper** [comment]
@AdoHe sure, I just said that I've had `--cert-dir` set at the time, and haven't tested otherwise. The error message is still pretty obscure.

**Comment by AdoHe** [comment]
@errordeveloper apiserver is designed to ignore `--cert-dir` if `--tls-cert-file` and `--tls-private-key-file` are provided.

**Comment by errordeveloper** [comment]
> apiserver is designed to ignore --cert-dir if --tls-cert-file and --tls-private-key-file are provided.

Sure. What I'm taking about is that the _error message is mega supper obscure_, when only `--tls-cert-file` is specified and `--tls-private-key-file` isn't.

**Comment by AdoHe** [comment]
@errordeveloper yeah, the error message should be updated to provide more clear info.

**Comment by lavalamp** [comment]
That is a super cryptic message

**Comment by errordeveloper** [comment]
`/lifecycle frozen`

**Comment by mk46** [comment]
I would like to give a try.
/assign
